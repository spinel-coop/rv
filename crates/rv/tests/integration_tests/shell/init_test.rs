use crate::common::RvTest;
use insta::assert_snapshot;

#[test]
fn test_shell_init_succeeds() {
    let test = RvTest::new();
    let output = test.rv(&["shell", "init", "zsh"]);
    output.assert_success();

    assert_snapshot!(output.normalized_stdout());
}

#[test]
fn test_nu_shell_init_succeeds() {
    let test = RvTest::new();
    let output = test.rv(&["shell", "init", "nu"]);
    output.assert_success();

    assert_snapshot!(output.normalized_stdout());
}

#[test]
fn test_shell_init_fails_without_shell() {
    let test = RvTest::new();
    let output = test.rv(&["shell", "init"]);
    output.assert_failure();

    assert_eq!(output.normalized_stdout(), "");
}
