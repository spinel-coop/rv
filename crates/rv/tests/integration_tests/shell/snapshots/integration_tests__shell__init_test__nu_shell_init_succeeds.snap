---
source: crates/rv/tests/integration_tests/shell/init_test.rs
expression: output.normalized_stdout()
---
$env.config = ($env.config | upsert hooks.env_change.PWD {
    [
        {
            |_, _| /tmp/bin/rv shell env nu | from json | load-env
        }
    ]
})
